Mathlib/Data/Matroid/Loop.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Circuit
+
+/-!
+# Matroid Loops
+A 'loop' of a matroid `M` is an element `e` satisfying one of the following equivalent conditions:
+* `e ∈ M.closure ∅`;
+* `{e}` is dependent in `M`;
+* `{e}` is a circuit of `M`;
+* no base of `M` contains `e`.
+In many mathematical contexts, loops can be thought of as 'trivial' or 'zero' elements;
+For linearly representable matroids, they correspond to the zero vector,
+and for graphic matroids, they correspond to edges incident with just one vertex (aka 'loops').
+As trivial as they are, loops can be created from matroids with no loops by taking minors or duals,
+so in many contexts it is unreasonable to simply forbid loops from appearing.
+This file defines loops, and provides API for interacting with them.
+
+# Main Declarations
+* For `M : Matroid α` and `e : α`, `M.IsLoop e` means that `e` is a loop of `M`,
+  defined as the statement `e ∈ M.closure ∅`.
+
+# Terminology
+The set `M.closure ∅ = {e | M.IsLoop e}` appears frequently in statements about loops.
+we stick with the more convenient spelling `M.closure ∅`,
+but refer to it in lemma names by the shorter term `loops`.
```

Comment:
Bearing this in mind, I wonder if it's any easier to define M.loops as this set, and use the spelling of being in that set for the predicate of being a loop. What do you think?

Reply:
I considered this. (I think) the reason I didn't do this for my repo is so that the set of loops was more transparently something I could apply closure lemmas to. But it looks like I have helper lemmas like `loops_subset_closure` to abstract these uses out anyway. So this is probably a good idea for saving keystrokes as well as clarity. I'l make the change.